### DISCUSSION TITLE
Does the Transactional API in the context of stream processing still require a consumer to assign to one specific partition?

### DISCUSSION THREAD

**Discussion by lecs** [confluentinc/confluent-kafka-python#1726] [Q&A]
Hi there, I'm looking at [this example of "how to perform a consume-transform-produce loop"](https://github.com/confluentinc/confluent-kafka-python/blob/master/examples/eos-transactions.py) - The comment in there states that "Prior to KIP-447 being supported each input partition requires its own transactional producer", but that was 4 years ago, and we're now post KIP-447, right? How would this example look nowadays? We can abstract away from partitions, right?

**Comment by pranavrth** [comment]
KIP-447 is supported now. The example is not updated though in this client.
